relaynet-cogrpc-jvm/src/main/java/tech/relaycorp/relaynet/cogrpc/client/CogRPCClient.kt

```diff
@@ -23,32 +25,49 @@ import tech.relaycorp.relaynet.cogrpc.readBytesAndClose
 import tech.relaycorp.relaynet.cogrpc.toCargoDelivery
 import tech.relaycorp.relaynet.cogrpc.toCargoDeliveryAck
 import java.io.InputStream
+import java.net.InetAddress
 import java.net.InetSocketAddress
 import java.net.URL
 import java.util.concurrent.TimeUnit
 import java.util.logging.Level
 import java.util.logging.Logger
 import kotlin.time.seconds
 
-class CogRPCClient
+open class CogRPCClient
 internal constructor(
     serverAddress: String,
-    val useTls: Boolean = true
+    val requireTls: Boolean = true
```

Comment:
I didn't realise this before: This flag shouldn't determine whether TLS is used; it should determine whether it's required. I'll enforce this a few lines below.

Reply:
My suggestion is to pass a channel builder through the constructor. It could be a ChannelBuilder interface or just a `(String) -> ManagedChannel`, and it could have a default value, to avoid changing code elsewhere. It would avoid opening the class and having a public var.